HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+the name owner MUST also pay the miner fee for this transaction. They do so by reducing
+the value of output 1, effectively paying the miner fee out of the money they earned
+by the sale.
+
+## Extra Notes
+
+### Bids must remain valid off-chain
+
+The partially-signed bid transactions are collected by the name owner and commit
+to spending a UTXO of some value V from the blockchain. In the simplest construction,
+nothing prevents the bidder from invalidating this off-chain transaction by
+double-spending the coin they used to bid. For this reason, the name owner must
+routinely check that all submitted bids are still valid. They can try finalizing
+the highest bid and if that fails, try to finalize the second-highest bid, etc.
+More advanced constructions may be possible to more effectively prevent this
+(lightning-network-style multisigs come to mind).
+
+### A semi-custodial platform could help ensure auction finality
+
+One possible scenario with this protocol is that the name owner never finalizes
+any bids. They might have decided a "reserve price" was not met or just changed
+their mind about the sale. One technique that may be used to ensure bidders that
+a winner will be chosen by a certain time is to include a trusted moderator.
+
+Auctions could be held on a platform like Shakedex or run by an agency with a
+good reputation like Namebase or Kyokan. This agent would publish a static public
+key for their platform, and name owners would have to incorporate that key along
+with a relative locktime into the HIP-0001 locking script like so:
+
+```
+OP_TYPE
+0x09 // TRANSFER
+OP_EQUAL
+OP_IF
+  OP_IF
+    <name-owner's public key>
+  OP_ELSE
+    <timespan of auction>
+    OP_CSV
+    OP_DROP
+    <platform's public key>
+  OP_ENDIF
+  OP_CHECKSIG
+OP_ELSE
+  OP_TYPE
+  0x0a // FINALIZE
+  OP_EQUAL
+OP_ENDIF
+```
+
+What this does is add a rule to the locking script such that, after a certain amount
+of time (i.e. 5 days) the platform operator can finalize the auction with their own
+private key instead of waiting for the name owner to do so. Technically this means that if
+the name owner does not finalize the auction in time (or simply transfer the name back to
+themselves -- publicly ending the auction on the blockchain) the platform can
+_steal the name OR the value of the highest bid_. In some cases, this may be the actual
+terms-of-service. In more common cases we can assume a platform with interest in
+maintaining a good reputation would award the highest bidder with the name and
+pay the name owner from output 1 of the final transaction.
+
+### Where & When
+
+Because this auction protocol is interactive, bidders need a place to send their
+partially-signed transactions where they are accessible to the name owner to finalize.
+A name owner could therefore open an auction by adding TXT records to their zone
+announcing the endpoint to send bids, along with the redeem script to prove the name
```

Comment:
i'm hoping we can introduce a HIP to cover this use case... working on that now

Reply:
Great!